### DISCUSSION TITLE
Second Storage Policy example doesn't seem to prevent users from accessing

### DISCUSSION THREAD

**Discussion by RiChrisMurphy** [supabase/supabase#2669] [Questions]
The storage policy example "Give users access to only their own a top level folder named as uid" does not prevent (local testing) a logged in user from inserting and downloading files that are not in their top level folder.

`CREATE POLICY "policy_name"
ON storage.objects FOR {operation} {USING | WITH CHECK} (
    -- restrict bucket
    bucket_id = {bucket_name}
    and auth.uid()::text = (storage.foldername(name))[1]
);`

Unsure if I'm missing anything, I made no alterations to the example.

**Comment by dshukertjr** [comment]
Hi @RiChrisMurphy!

Maybe this was implied and you actually did this, but you can separate the example into individual `insert`, `update` or `delete`. Example for `insert` would be the following. Note that I have deleted the part about restricting `bucket_id`, but you can add it inn if you would like that!
```sql
CREATE POLICY "policy_name" ON storage.objects FOR insert WITH CHECK ( auth.uid()::text = (storage.foldername(name))[1] );
```

P.S.
I was not able to find "Give users access to only their own a top level folder named as uid" example, but where did you find it? Is it on the official Supabase website? I couldn't find it anywhere and it sounds like a cool page to book mark 😉
